src/Sparrow/Allocators.PointerTypes.cs

```diff
@@ -269,21 +269,17 @@ public Span<byte> AsSpan()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public Span<byte> AsSpan(int length)
         {
-#if VALIDATE
             if (length > Size)
                 throw new ArgumentException($"{nameof(length)} cannot be bigger than block size.");            
-#endif
 
             return new Span<byte>(Ptr, length);
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public Span<byte> AsSpan(int start, int length)
         {
-#if VALIDATE
             if (start + length > Size)
                 throw new ArgumentException($"{nameof(length)} cannot be bigger than block size.");            
```

Comment:
This JIT is now smart enough to move the exception throwing code to a cold location without creating dedicated method?